tests/functional/test_package.py

```diff
@@ -23,6 +34,589 @@ def index():
+        directory = os.path.join(*prefix)
+        filename_without_ext = filename[:-4]
+        pkg_name, pkg_version = filename_without_ext.split('-')
+        builder = FakeSdistBuilder()
+        builder.write_fake_sdist(directory, pkg_name, pkg_version)
+
+    def execute(self, args):
+        """Generate the file in the target_dir."""
+        if self._dirarg:
+            target_dir = None
+            for i, arg in enumerate(args):
+                if arg == self._dirarg:
+                    target_dir = args[i+1]
+            if target_dir:
+                filepath = os.path.join(target_dir, self._filename)
+                if filepath.endswith('.whl'):
+                    self._build_fake_whl(target_dir, self._filename)
+                else:
+                    self._build_fake_sdist(filepath)
+
+
+@pytest.fixture
+def osutils():
+    return OSUtils()
+
+
+@pytest.fixture
+def pip_runner():
+    pip = FakePip()
+    pip_runner = PipRunner(pip)
+    return pip, pip_runner
+
+
+class TestDependencyBuilder(object):
+    def _write_requirements_txt(self, packages, directory):
+        contents = '\n'.join(packages)
+        filepath = os.path.join(directory, 'requirements.txt')
+        with open(filepath, 'w') as f:
+            f.write(contents)
+
+    def _make_appdir_and_dependency_builder(self, reqs, tmpdir, runner):
+        appdir = str(_create_app_structure(tmpdir))
+        self._write_requirements_txt(reqs, appdir)
+        builder = DependencyBuilder(OSUtils(), runner)
+        return appdir, builder
+
+    def test_can_get_whls_all_manylinux(self, tmpdir, pip_runner):
+        reqs = ['foo', 'bar']
+        pip, runner = pip_runner
+        appdir, builder = self._make_appdir_and_dependency_builder(
+            reqs, tmpdir, runner)
+        requirements_file = os.path.join(appdir, 'requirements.txt')
+        pip.packages_to_download(
+            expected_args=['-r', requirements_file, '--dest', mock.ANY],
+            packages=[
+                'foo-1.2-cp36-cp36m-manylinux1_x86_64.whl',
+                'bar-1.2-cp36-cp36m-manylinux1_x86_64.whl'
+            ]
+        )
+
+        builder.build_site_packages(appdir)
+        site_packages = builder.site_package_dir(appdir)
+        installed_packages = os.listdir(site_packages)
+
+        pip.validate()
+        for req in reqs:
+            assert req in installed_packages
+
+    def test_can_expand_purelib_whl(self, tmpdir, pip_runner):
+        reqs = ['foo']
+        pip, runner = pip_runner
+        appdir, builder = self._make_appdir_and_dependency_builder(
+            reqs, tmpdir, runner)
+        requirements_file = os.path.join(appdir, 'requirements.txt')
+        pip.packages_to_download(
+            expected_args=['-r', requirements_file, '--dest', mock.ANY],
+            packages=[
+                'foo-1.2-cp36-cp36m-manylinux1_x86_64.whl'
+            ],
+            whl_contents=['{data_dir}/purelib/foo/placeholder']
```

Comment:
Offline discussion: Those were there for the implicit case where I do not supply wheel contents and I need to calculate the data directory name anyway so I reused that mechanism. Its clearer to write it out in the explicit tests so I will change it to do that rather than rely on the format string.